src/App.tsx

```diff
@@ -1,51 +1,89 @@
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
-};
+interface State {
+  selectedGoods: string[],
+}
+
+export class App extends React.Component <{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title title">
+            { selectedGoods.length === 0
+              ? 'No goods selected'
+              : (`${selectedGoods.join(', ')} ${selectedGoods.length === 1
+                ? 'is'
+                : 'are'
+              } selected`)}
+          </h1>
+
+          {selectedGoods.length > 0 && (
+            <button
+              type="button"
+              className="
+                App__clear
+                button
+                is-small"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+            >
+
+              Clear
+            </button>
+          )}
+        </header>
+
+        <ul className="container">
+          {goodsFromServer.map(good => (
+            <li className={classNames(
+              'Good',
+              selectedGoods.includes(good)
+                ? 'Good--active'
+                : '',
+            )}
+            >
+
+              {good}
+
+              <button
+                type="button"
+                className={`Good__${selectedGoods.includes(good)
+                  ? 'remove is-danger is-outlined'
+                  : 'select is-success'}
+                  button is-small`}
+                onClick={() => (
+                  selectedGoods.includes(good)
+                    ? (selectedGoods.splice(selectedGoods.indexOf(good), 1)
+                      && this.setState({
+                        selectedGoods,
+                      })
+                    )
+                    : (selectedGoods.push(good)
+                        && this.setState({
+                          selectedGoods,
+                        })
+                    )
+                )}
```

Comment:
splice and push is a mutable method, u can't mutate state, let's use the filter. And move to method